solutions/classes/input-reader.js

```diff
@@ -0,0 +1,38 @@
+let fs = require("fs");
+let data = fs.readFileSync(0, "utf-8");
+let idx = 0;
+data = data.split("\n");
+
+function readLine() {
+  idx++;
+  return data[idx - 1];
+}
+
+// -------- Do NOT edit anything above this line ----------
+const Inputreader = class {
+  Data = readLine();
+  constructor(Index) {
+    this.Index = Index;
+  }
+  read_strings() {
+    console.log(this.Index + " " + this.Data);
+  }
+  read_integers() {
+    console.log(this.Index + " " + parseInt(this.Data));
+  }
+  read_floats() {
+    console.log(this.Index + " " + parseFloat(this.Data).toFixed(2));
+  }
+};
+
+var Num = parseInt(readLine());
+var typeofinput = readLine();
+for (let i = 0; i < Num; i++) {
+  if (typeofinput === "string") {
+    new Inputreader(i).read_strings();
```

Comment:
I think its better to store the instance of the class as a class variable first, and then use it everytime.
like `let NewInputReader = new InputReader(i)`. Please correct me if its not a good practice of storing as variables.

Reply:
> 

@sailesh-ramayanam  should i make the above change in the code?